### PR TITLE
feat: simple merkle tree

### PR DESCRIPTION
**Pull request by ratankaliani**
## Overview
- Add simple merkle tree.
- Clean up existing Tendermint merkle tree.

### CODE REVIEW

File: plonky2x/core/src/frontend/merkle/tendermint.rs [hunk 1]
```diff
@@ -8,13 +9,58 @@ use crate::frontend::merkle::utils::log2_ceil_usize;
...
+        &mut self,
+        leaves: ArrayVariable<BytesVariable<LEAF_SIZE_BYTES>, MAX_NB_LEAVES>,
+        nb_enabled_leaves: Variable,
+    ) -> Bytes32Variable;
+}
+
 /// Merkle Tree implementation for the Tendermint spec (follows Comet BFT Simple Merkle Tree spec: https://docs.cometbft.com/main/spec/core/encoding#merkle-trees).
+/// Adds pre-image prefix of 0x01 to inner nodes and 0x00 to leaf nodes for second pre-image resistance.
+/// Computed root hash is independent of the number of empty leaves, unlike the simple Merkle Tree.
 /// TODO: Create generic interface for Merkle trees to implement.
```
Comment (puma314) [inline plonky2x/core/src/frontend/merkle/tendermint.rs:60]: nit: should we remove this comment? it seems like you went with 2 separate traits?

File: plonky2x/core/src/frontend/merkle/tendermint.rs [hunk 2]
```diff
@@ -125,53 +163,48 @@ impl<L: PlonkParameters<D>, const D: usize> CircuitBuilder<L, D> {
...
         // Whether to treat the validator as empty.
         // Pad the enabled array to be a power of 2.
-        let mut current_node_enabled = leaves_enabled.clone();
-        current_node_enabled.resize(padded_nb_leaves, self._false());
-
-        let mut merkle_layer_size = padded_nb_leaves;
+        let mut current_node_enabled = Vec::new();
+        for i in 0..padded_nb_leaves {
+            let idx = self.constant::<Variable>(L::Field::from_canonical_usize(i));
+            current_node_enabled.push(self.lt(idx, nb_enabled_leaves));
```
Comment (puma314) [inline plonky2x/core/src/frontend/merkle/tendermint.rs:187]: this could be expensive, you should check if it pushes you over the power of 2

**Review by puma314 — APPROVED** [review]

File: plonky2x/core/src/frontend/merkle/tendermint.rs (hunk 2 above)
Comment (ratankaliani) [reply plonky2x/core/src/frontend/merkle/tendermint.rs:187]: From some quick benchmarks, this doesn't seem like a significant issue. Going to leave this for now.

Comment (ratankaliani) [reply plonky2x/core/src/frontend/merkle/tendermint.rs:187]: We can add this optimization in the future if necessary.

Comment (ratankaliani) [reply plonky2x/core/src/frontend/merkle/tendermint.rs:187]: Actually nvm, going to add now.

**Pull request merged**